### DISCUSSION TITLE
Trace methods not getters

### DISCUSSION THREAD

**Discussion by ghost** [btraceio/btrace#553] [Q&A]
I realize Java doesn't have properties like c# does, but still, it's common to have the equivalent construct; lots of codebases have getX/setX w/o anything interesting happening there.

Is there a way to filter such methods out?
Or is there a way to express the class/method filters with something more manageable than a regex, something like a file?

Thanks!

**Comment by ghost** [comment]
@jbachorik any idea?

**Reply by jbachorik** [reply]
You can use any valid regex in the method definition so you could do something like
`@OnMethod(clazz = "MyClass" method = "/^(?!(get|set)).*$/" ...)`  (see https://stackoverflow.com/questions/406230/regular-expression-to-match-a-line-that-doesnt-contain-a-word)

I haven't tried this yet but it should work with the caveat of matching anything starting with `get` or `set` prefix.

**Reply by ghost** [reply]
yeah, i tried that and it does work, but it cuts out too many useful methods.
What would work really well if we could specify a discrete list of methods to include/exclude in a csv on disk, or smth like that.

**Reply by jbachorik** [reply]
You could use probe mappings for that - see https://github.com/btraceio/btrace/blob/develop/btrace-dist/src/main/resources/samples/java.net.socket.xml - and then use `@OnProbe` annotation to assign handlers to those mappings.

**Reply by omkar-ti** [reply]
if I have a handler for a probe mapping I created and  I also have a handler for @onMethod in my script, what will happen if a method satisfies conditions for both?

**Reply by jbachorik** [reply]
Both `@OnProbe` and `@OnMethod` will be applied and the handlers should be called. The order in which they will be called is undefined.
